blocks/search/search.js

```diff
@@ -104,172 +100,128 @@ function addSearchByListeners(wrapper, form) {
 function populateFilter(select, items) {
   const docRange = document.createRange();
   let htmlFragment = '';
+  const urlParams = new URLSearchParams(window.location.search);
   items.forEach((item) => {
+    const itemValue = urlParams.get('cat') ? item.toLowerCase() : item;
     htmlFragment += `
-      <option value="${item.toLowerCase()}">${item}</option>
+      <option value="${itemValue}">${item}</option>
     `;
   });
   const fragment = docRange.createContextualFragment(htmlFragment);
   select.appendChild(fragment);
 }
 
+export const getAndApplySearchResults = async ({ isFirstSet }) => {
+  const { MAX_PRODUCTS_PER_QUERY } = SEARCH_CONFIG;
```

Comment:
Is it possible to break these 100 something lines of code into functional parts that we can understand?
This is s bunch of dozens of variables being set one after the other with actions on the middle of them without any kind of explained logic or clear sense. 
This makes this extremely difficult to understand and maintain...

Reply:
I extracted several functions to make the main one more declarative, I hope that helps